pygrocy/grocy_api_client.py

```diff
@@ -394,6 +400,40 @@ def consume_product(
 
         self._do_post_request(f"stock/products/{product_id}/consume", data)
 
+    def add_product_by_barcode(
+        self,
+        barcode: str,
+        amount: float,
+        price: float,
+        best_before_date: datetime = None,
+        transaction_type: TransactionType = TransactionType.PURCHASE,
+    ):
+        data = {
+            "amount": amount,
+            "transaction_type": transaction_type.value,
+            "price": price,
+        }
+
+        if best_before_date is not None:
+            data["best_before_date"] = best_before_date.strftime("%Y-%m-%d")
+
+        return self._do_post_request(f"stock/products/by-barcode/{barcode}/add", data)
+
+    def consume_product_by_barcode(
+        self,
+        barcode: str,
+        amount: float = 1,
+        spoiled: bool = False,
+        transaction_type: TransactionType = TransactionType.CONSUME,
```

Comment:
can there be a transaction_type different from `CONSUME`?

Reply:
Not for this method - will always be consume

Reply:
Ok, so I guess we do not need that parameter, could you remove it please?

Reply:
Done :)